extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/TenantConfigBean.java

```diff
@@ -36,23 +40,29 @@ public TenantConfigBean(
     }
 
     public Uni<TenantConfigContext> createTenantContext(OidcTenantConfig oidcConfig, boolean dynamicTenant) {
-        if (oidcConfig.logout.backchannel.path.isPresent()) {
+        var tenantId = oidcConfig.getTenantId().orElseThrow();
+        if (dynamicTenant && oidcConfig.logout.backchannel.path.isPresent()) {
             throw new ConfigurationException(
-                    "BackChannel Logout is currently not supported for dynamic tenants");
+                    "BackChannel Logout is currently not supported for dynamic tenants (tenant ID: " + tenantId + ")");
         }
-        var tenantId = oidcConfig.getTenantId().orElseThrow();
-        if (!dynamicTenantsConfig.containsKey(tenantId)) {
+        var tenants = dynamicTenant ? dynamicTenantsConfig : staticTenantsConfig;
+        var tenant = tenants.get(tenantId);
+        if (tenant == null || !tenant.ready) {
```

Comment:
I don't recall right now, but I thought it was only static tenants that can be in a non-ready state. Dynamic tenants are created on demand, they are either ready or not. Please double check

Reply:
Yes, the ready flag can only be false for static tenants - since it is only with static tenants that we can have a situation where the provider is not available at the startup, so we can retry later at the first request time.
But dynamic tenants are created at the request time, and they are only supposed to be recorded if they are ready.